lib/logthrdest/tests/test_logthrdestdrv.c

```diff
@@ -771,6 +775,25 @@ Test(logthrdestdrv, test_explicit_ack_accept)
   cr_assert(dd->super.shared_seq_num == 11, "%d", dd->super.shared_seq_num);
 }
 
+Test(logthrdestdrv, test_if_generate_persist_name_called_via_the_logpipe_interface)
+{
+  cr_assert(dd->generate_persist_name_call_counter == 1,
```

Comment:
The test should not limit a code to call generate persist name only once. As I think the important thing is that it is called at least once. The fact that it is cached, and only called once is an implementation detail.
On the other hand, I do not see a test that checks regarding the persist-name actually correctly set.